mlflow/deployments/base.py

```diff
@@ -70,6 +70,226 @@ def target_help():
+
+        :param deployment: Name of deployment to delete
+        :param endpoint: Name of the Endpoint containing the deployment to delete
+        :return: None
+        """
+        pass
+
+    @abc.abstractmethod
+    def list_deployments(self, endpoint):
+        """
+        List deployments within the specified endpoint. This method is expected to return
+        an unpaginated list of all deployments (an alternative would be to return a dict
+        with a 'deployments' field containing the actual deployments, with plugins able to
+        specify other fields, e.g. a next_page_token field, in the returned dictionary for
+        pagination, and to accept a `pagination_args` argument to this method for passing
+        pagination-related args).
+
+        :param endpoint: Name of the Endpoint to list all contained deployments
+        :return: A list of dicts corresponding to deployments. Each dict is guaranteed to
+                 contain a 'name' key containing the deployment name. The other fields of
+                 the returned dictionary and their types may vary across deployment targets.
+        """
+        pass
+
+    @abc.abstractmethod
+    def get_deployment(self, deployment, endpoint):
+        """
+        Returns a dictionary describing the specified deployment, throwing a
+        py:class:`mlflow.exception.MlflowException` if no deployment exists with the provided
+        name in the provided endpoint.
+        The dict is guaranteed to contain an 'name' key containing the deployment name.
+        The other fields of the returned dictionary and their types may vary across
+        deployment targets.
+
+        :param deployment: Name of deployment to fetch
+        :param endpoint: Name of the Endpoint containing the deployment to fetch
+        """
+        pass
+
+    @abc.abstractmethod
+    def predict(self, endpoint, df, deployment=None):
+        """
+        Compute predictions on the pandas DataFrame ``df`` using the specified endpoint.
+        Note that the input/output types of this method matches that of `mlflow pyfunc predict`
+        (we accept a pandas DataFrame as input and return either a pandas DataFrame,
+        pandas Series, or numpy array as output).
+
+        By default, the prediction will be issued to the endpoint and subject to any traffic
+        routing that is in place on the endpoint. Alternatively, the name of a specific
+        deployment within the endpoint can be provided in order to issue a prediction against
+        that specific deployment, for the sake of testing.
+
+        :param endpoint: Name of the endpoint to issue the prediction against
+        :param df: Pandas DataFrame to use for inference
+        :param deployment: If provided issue a prediction against the specified endpoint,
+                           bypassing the endpoint traffic settings
+        :return: A pandas DataFrame, pandas Series, or numpy array
+        """
+        pass
+
+    @experimental
```

Comment:
Is explain method strictly required?

Reply:
This was another one that existed in the BaseDeploymentClient, so I added it here as well. I think it's probably good to maintain that parity.

Reply:
Not currently, possibly in the future. But given we decided that any endpoint related deployment operations should go through the endpoint client, I'm assuming we want to match the current deployment client for existing methods

Reply:
It's much easier to add the method back than taking it away. Shall we file another PR when the need arises? We might have a different idea by then on the API (since the requirements are much more concrete). WDYT?